xskillscore/tests/test_probabilistic.py

```diff
@@ -476,3 +478,225 @@ def test_rps_perfect_values(o, category_edges):
+        dim="time",
+        return_results="area",
+    )
+    sk_area = roc_auc_score(observation_1d_long > 0, forecast_1d_long * 0)
+    np.testing.assert_allclose(xs_area, sk_area)
+
+
+@pytest.mark.parametrize("drop_intermediate", [False, True])
+def test_roc_bin_edges_continuous_against_sklearn(
+    forecast_1d_long, observation_1d_long, drop_intermediate
+):
+    """Test xs.roc against sklearn.metrics.roc_curve/auc_score."""
+    fp = np.clip(forecast_1d_long, 0, 1)  # prob
+    ob = observation_1d_long > 0  # binary
+    # sklearn
+    sk_fpr, sk_tpr, _ = roc_curve(ob, fp, drop_intermediate=drop_intermediate)
+    sk_area = roc_auc_score(ob, fp)
+    # xs
+    xs_fpr, xs_tpr, xs_area = roc(
+        ob,
+        fp,
+        "continuous",
+        drop_intermediate=drop_intermediate,
+        return_results="all_as_tuple",
+    )
+    np.testing.assert_allclose(xs_area, sk_area)
+    if not drop_intermediate:  # drops sometimes one too much or too little
+        assert (xs_fpr == sk_fpr).all()
+        assert (xs_tpr == sk_tpr).all()
+
+
+def test_roc_bin_edges_drop_intermediate(forecast_1d_long, observation_1d_long):
+    """Test that drop_intermediate reduces probability_bins in xs.roc ."""
+    fp = np.clip(forecast_1d_long, 0, 1)  # prob
+    ob = observation_1d_long > 0  # binary
+    # xs
+    txs_fpr, txs_tpr, txs_area = roc(
+        ob, fp, "continuous", drop_intermediate=True, return_results="all_as_tuple"
+    )
+    fxs_fpr, fxs_tpr, fxs_area = roc(
+        ob, fp, "continuous", drop_intermediate=False, return_results="all_as_tuple"
+    )
+    # same area
+    np.testing.assert_allclose(fxs_area, txs_area)
+    # same or less probability_bins
+    assert len(fxs_fpr) >= len(txs_fpr)
+    assert len(fxs_tpr) >= len(txs_tpr)
+
+
+def test_roc_keeps_probability_bin_as_coord(
+    observation_1d_long, forecast_1d_long, symmetric_edges
+):
+    """Test that roc keeps probability_bin as coords."""
+    fpr, tpr, area = roc(
+        observation_1d_long,
+        forecast_1d_long,
+        symmetric_edges,
+        drop_intermediate=False,
+        return_results="all_as_tuple",
+    )
+    assert (tpr.probability_bin == symmetric_edges).all()
+    assert (fpr.probability_bin == symmetric_edges).all()
+
+
+def test_roc_bin_edges_symmetric(
+    observation_1d_long, forecast_1d_long, symmetric_edges
+):
+    """Test that roc bin_edges works increasing or decreasing order."""
+    fpr, tpr, area = roc(
+        observation_1d_long,
+        forecast_1d_long,
+        symmetric_edges,
+        drop_intermediate=False,
+        return_results="all_as_tuple",
+    )
+    fpr2, tpr2, area2 = roc(
+        observation_1d_long,
+        forecast_1d_long,
+        symmetric_edges[::-1],
+        drop_intermediate=False,
+        return_results="all_as_tuple",
+    )
+    assert_identical(fpr, fpr2.sortby(fpr.probability_bin))
+    assert_identical(tpr, tpr2.sortby(tpr.probability_bin))
+    assert_identical(area, area2)
```

Comment:
should this be the case? or rather `area2=1-area`, then no `np.sort()` ascending on `bin_edges`

Reply:
@dougiesquire may have some thoughts on this.

Reply:
What you have here makes sense to me, but I'm not sure I fully understand the question. I think the area should be the same for increasing or decreasing `bin_edges`, which seems to be what you're testing.